### DISCUSSION TITLE
Does pyenv use system's `GCC`?

### DISCUSSION THREAD

**Discussion by ego-thales** [pyenv/pyenv#2758] [Q&A]
Hi,

I would juste like to understand how `pyenv` installs new versions of python when I run for example `pyenv install 3.11`. Does it compile the appropriate binaries with my local `GCC`? Or does it simply download a preinstalled version?

I'm not very familiar with compilation of python and DLL's, so feel free to input the appropriate information for me to understand a bit better :)

Thanks!

**Comment by native-api** [comment]
Most of the build scripts download a source package and build it.
So it would use whatever the package's build logic uses -- which is typically your default system compiler.

You can run `pyenv install -v <version>` to see the build's output, or even `PYENV_DEBUG=1 <pyenv command>` to get a full trace of execution (lots of output, so better redirect it to a log file to view with `less`).

**Comment by ego-thales** [comment]
Thanks for your answer.
I came across [this](https://github.com/pyenv/pyenv/blob/master/plugins/python-build/README.md#special-environment-variables) (python-build README), explaining that `pyenv` uses the `CC` environment variable as C compiler. If not set by the user, python-build does look for it on your system, via the [`locate_gcc`](https://github.com/pyenv/pyenv/blob/79a501139fd106b39c6095467930ac506822c4c5/plugins/python-build/bin/python-build#L1211-L1259) function.
